File: Java/01/tasks-manager-application/src/main/java/org/tasksmanager/Repositories/DoneTasksRepository.java
package org.tasksmanager.Repositories;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;
import org.cqrs.Repository;


    public class DoneTasksRepository implements Repository<DoneTaskDao>
    {
        private static ConcurrentHashMap<UUID, DoneTaskDao> _storage = new ConcurrentHashMap<UUID, DoneTaskDao>();

        public void Delete(UUID id)
        {
            if (_storage.containsKey(id)) _storage.remove(id);
        }

        public List<DoneTaskDao> GetAll()
        {
            return (List<DoneTaskDao>) _storage.elements();
        }

        public DoneTaskDao GetById(UUID id)
        {
            if (_storage.containsKey(id)) return _storage.get(id);
            return null;
        }

        public DoneTaskDao Save(DoneTaskDao item)
        {
            _storage.put(item.getId(), item);
            return item;
        }
    }

File: Net/02/TasksManager.Application/Repositories/DoneTasksRepository.cs
using Cqrs;
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;

namespace TasksManager.Repositories
{
    public class DoneTasksRepository : IRepository<DoneTaskDao, Guid>
    {
        private static ConcurrentDictionary<Guid, DoneTaskDao> _storage = new ConcurrentDictionary<Guid, DoneTaskDao>();

        public void Delete(Guid id)
        {
            DoneTaskDao deleted;
            if (_storage.ContainsKey(id)) _storage.TryRemove(id, out deleted);
        }

        public List<DoneTaskDao> GetAll()
        {
            return _storage.Values.ToList();
        }

        public DoneTaskDao GetById(Guid id)
        {
            if (_storage.ContainsKey(id)) return _storage[id];
            return null;
        }

        public DoneTaskDao Save(DoneTaskDao item)
        {
            _storage[item.Id] = item;
            return item;
        }
    }
}
